### ISSUE TITLE
Robolectric is incorrectly specified as non-test dependencies

### ISSUE THREAD

**Issue by sebastienlabine** [facebook/facebook-android-sdk#1117]
### Checklist before submitting a bug report

- [X] I've updated to the latest released version of the SDK
- [X] I've searched for existing [Github issues](https://github.com/facebook/facebook-android-sdk/issues)
- [X] I've looked for existing answers on [Stack Overflow](https://facebook.stackoverflow.com), the [Facebook Developer Community Forum](https://developers.facebook.com/community/) and the [Facebook Developers Group](https://www.facebook.com/groups/fbdevelopers)
- [X] I've read the [Code of Conduct](CODE_OF_CONDUCT.md)
- [X] This issue is not security related and can safely be disclosed publicly on GitHub

### Java version

17.0.4.1

### Android version

API 32, Android 12

### Android SDK version

15.+

### Installation platform & version

Gradle 7.3.3

### Package

Gaming Services

### Goals

I want to be able to build my solution without having to manually resolve the dependency of bouncycastle:bcprov.

### Expected results

Robolectric as a test dependency https://github.com/facebook/facebook-android-sdk/blob/0970b6a75057b5d243c58e34c17455e896042925/facebook-gamingservices/build.gradle.kts#L37 as stated here https://github.com/thebergamo/react-native-fbsdk-next/issues/323

This would result in a build that doesn't fail because of this dependency.

### Actual results

Building tasks of modules that are also using a different version of bouncycastle (used by Robolectric) make the whole build fail.
```
Duplicate class org.bouncycastle.x509.util.StreamParsingException found in modules jetified-bcprov-jdk15on-1.57 (org.bouncycastle:bcprov-jdk15on:1.57) and jetified-bcprov-jdk15to18-1.70 (org.bouncycastle:bcprov-jdk15to18:1.70)
```

The current solution to bypass this problem is to manually resolve the dependency of bouncycastle to org.bouncycastle:bcprov-jdk15to18:1.70 by doing 
```
android {
configurations.all {
    c -> c.resolutionStrategy.eachDependency {
        DependencyResolveDetails dependency ->
            println dependency.requested.group
             if (dependency.requested.group == 'org.bouncycastle') {
                dependency.useTarget 'org.bouncycastle:bcprov-jdk15to18:1.68'
            }
    }
}
}
```
as referered here https://github.com/stripe/stripe-android/issues/3173

### Steps to reproduce

1 - Create a project.
2 - In your build.gradle file, implement your SDK and org.bouncycastle:bcutil-jdk15to18:1.70
3 - Try to build

**Comment by isamotiuc** [comment]
I wasted half a day trying to understand why my release APK increased size +9mb and why ICU4J was added in the build :(

**Comment by mingcaozhang** [comment]
Apologies for this, issue is now fixed in release [15.0.2](https://github.com/facebook/facebook-android-sdk/releases/tag/sdk-version-15.0.2).
